src/charm.py

```diff
@@ -60,6 +64,27 @@ def __init__(self, *args):
             self.unit.get_container(CONTAINER), User(name=PEBBLE_USER[0], group=PEBBLE_USER[1])
         )
 
+        # oauth
+        self.auth_proxy_relation_name = "auth-proxy"
+
+        self.auth_proxy = AuthProxyRequirer(
+            self, self.auth_proxy_config, self.auth_proxy_relation_name
+        )
+
```

Comment:
You could observe `auth_proxy_relation_removed` event to inform that the app is not protected anymore.

Reply:
Hi @natalian98 I added this but I got an error when I try to remove the relation between the charm and oathkeeper ( I think it is related to this [issue](https://github.com/canonical/oathkeeper-operator/issues/44))

Reply:
Sorry for that, I just merged a PR with a fix. Could you fetch the latest lib?

Reply:
Done